src/main/scala/analysis/IDEAnalysis.scala

```diff
@@ -0,0 +1,55 @@
+package analysis
+
+import ir.{CFGPosition, Command, DirectCall, GoTo, Procedure, Program}
+
+/**
+ * Base trait for IDE analyses.
+ * @tparam D the type of items
+ * @tparam L the type of the value lattice
+ * Adapted from Tip
+ * https://github.com/cs-au-dk/TIP/blob/master/src/tip/solvers/IDEAnalysis.scala
+ * The special item representing the empty element in IDE.
+ */
+final case class Lambda()
+
+trait IDEAnalysis[E, EE, C, R, D, T, L <: Lattice[T]] {
+  val program: Program
+
+  type DL = Either[D, Lambda]
+
+  /**
+   * The value lattice.
+   */
+  val valuelattice: L
+
+  /**
+   * The edge lattice.
+   */
+  val edgelattice: EdgeFunctionLattice[T, valuelattice.type]
+
+  /**
+   * Edges for call-to-entry.
+   */
+  def edgesCallToEntry(call: C, entry: E)(d: DL): Map[DL, edgelattice.Element]
+
+  /**
+   * Edges for exit-to-aftercall.
+   */
+  def edgesExitToAfterCall(exit: EE, aftercall: R)(d: DL): Map[DL, edgelattice.Element]
```

Comment:
This should be named edgesReturnToAfterCall given that we don't use TIP's concept of a 'function exit'?

Reply:
I think edgesExitToAfterCall makes more sense since the exit can be a return (indirect call) or a procedure depending on whether the analysis is forward or not. Exit just refers to the CFGPosition where the analysis exits the function call. 
Similar to edgesCallToEntry where entry can also be a return (indirect call) or a procedure.